crypto/o_fopen.c

```diff
@@ -15,6 +15,9 @@
 # ifdef _WIN32
 #  include <windows.h>
 # endif
+# ifdef __DJGPP__
+#  include <unistd.h>
```

Comment:
Should we just take away that config parameter?

Reply:
I don't follow... Do you mean unistd (that is customarily assigned to <unistd.h>)? If yes, then in long run, yes, totally. Even though there might be kind of a chicken-n-egg problem. I mean if you want make it #ifdef conditional, then you you'd probably want to use POSIX_something, but those customarily come with unistd.h. Well, it's not like they are defined in unistd.h, but unistd.h would be number 1 choice to get them without thinking too much about what system is it. Though there is one predefine that is quite reliable, `__unix__`/`__unix`... In case you wonder, if it's defined by djgpp compiler? Yes, Cygwin? Yes. Can you guess if "MSYS" compiler does? "MinGW" cross-compiler? Answers are of course "yes" and "no". This by the way reliable way to tell apars "native" and "hosted" that are discussed in updated NOTES.WIN...

But in either case *here*, in o_fopen.c it would still be DJGPP-specific. In normal case this module would require stdio along. This is non-POSIX thing. And it's only DJGPP that needs that POSIX-y pathconf to take a decision.

Reply:
> Though there is one predefine that is quite reliable, __unix__/__unix... In case you wonder,

Well, there is one that doesn't follow the pattern. Can you guess which one? Vendor that always has to be special... Anybody?

Reply:
I should mark my side notes explicitely as such...  But to answer your question, mine was about the config attribute `unistd`, which is usually assigned `"<unistd.h>"`, except on nextstep...

Reply:
So many choices... MSFT... Apple ...  don't keep us guessing, @dot-asm

Reply:
I meant can you guess one that you would *totally* expect to pre-define `__unix__` in compiler. One that is equipped with Bourne shell, has find, grep, ls, ... and even has <unistd.h> that defines POSIX_things... It can't be Windows... But of course it's Apple, you guessed it.